hibernate-core/src/main/java/org/hibernate/dialect/DatabaseVersion.java

```diff
@@ -0,0 +1,253 @@
+
+		return major >= otherMajor;
+	}
+
+	/**
+	 * {@link #isSame} or {@link #isAfter}
+	 */
+	default boolean isSince(int otherMajor, int otherMinor) {
+		final int major = getDatabaseMajorVersion();
+		final int minor = getDatabaseMinorVersion();
+
+		return major > otherMajor
+				|| ( major == otherMajor && minor >= otherMinor );
+	}
+
+	/**
+	 * {@link #isSame} or {@link #isAfter}
+	 */
+	default boolean isSince(int otherMajor, int otherMinor, int otherMicro) {
+		final int major = getDatabaseMajorVersion();
+		final int minor = getDatabaseMinorVersion();
+		final int micro = getDatabaseMicroVersion();
+
+		return major > otherMajor
+				|| ( major == otherMajor && minor > otherMinor )
+				|| ( major == otherMajor && minor == otherMinor && micro >= otherMicro );
+	}
+
+	/**
+	 * Determine whether this version comes after the passed one
+	 */
+	default boolean isAfter(DatabaseVersion other) {
+		return isAfter( other.getDatabaseMajorVersion(), other.getDatabaseMinorVersion() );
+	}
+
+	/**
+	 * Determine whether this version after the passed one
+	 */
+	default boolean isAfter(Integer major, Integer minor) {
+		return isAfter( (int) major, minor == null ? NO_VERSION : minor );
+	}
+
+	/**
+	 * Determine whether this version after the passed one
+	 */
+	default boolean isAfter(int major) {
+		return getDatabaseMajorVersion() > major;
+	}
+
+	/**
+	 * Determine whether this version after the passed one
+	 */
+	default boolean isAfter(int major, int minor) {
+		return getDatabaseMajorVersion() > major
+				|| ( getDatabaseMajorVersion() == major && getDatabaseMinorVersion() > minor );
+	}
+
+	/**
+	 * Determine whether this version after the passed one
+	 */
+	default boolean isAfter(int otherMajor, int otherMinor, int otherMicro) {
+		final int major = getDatabaseMajorVersion();
+		final int minor = getDatabaseMinorVersion();
+		final int micro = getDatabaseMicroVersion();
+
+		return major > otherMajor
+				|| ( major == otherMajor && minor > otherMinor )
+				|| ( major == otherMajor && minor == otherMinor && otherMicro > micro );
+	}
+
+	/**
+	 * Determine whether this version comes before the passed one
+	 */
+	default boolean isBefore(DatabaseVersion other) {
+		return isBefore( other.getDatabaseMajorVersion(), other.getDatabaseMinorVersion() );
+	}
+
+	/**
+	 * Determine whether this version before the passed one
+	 */
+	default boolean isBefore(int major, int minor) {
+		return ! isSince( major, minor );
+	}
+
+	/**
+	 * Determine whether this version before the passed one
+	 */
+	default boolean isBefore(int major) {
+		return ! isSince( major );
+	}
+
+	/**
+	 * Determine whether this version before the passed one
+	 */
+	default boolean isBefore(Integer major, Integer minor) {
+		return isBefore( (int) major, minor == null ? NO_VERSION : minor );
+	}
+
+	/**
+	 * Determine whether this version before the passed one
+	 */
+	default boolean isBefore(int otherMajor, int otherMinor, int otherMicro) {
```

Comment:
I would prefer the name `isAtLeast` but YMMV

Reply:
As an alternative to isBefore?  I assume you mean isSince?

Reply:
Yeah. Apparently I clicked on the wrong line

Reply:
I landed on `#isSameOrAfter`